OpenRA.Mods.Common/Activities/Parachute.cs

```diff
@@ -0,0 +1,96 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+	public class Parachute : Activity
+	{
+		WPos dropPosition;
+		WPos currentPosition;
+		WVec fallVector;
+		bool triggered = false;
+
+		public Parachute(Actor self, WPos dropPosition)
+		{
+			this.dropPosition = dropPosition;
+		}
+
+		Activity FirstTick(Actor self)
+		{
+			triggered = true;
+
+			var um = self.TraitOrDefault<UpgradeManager>();
+			if (um != null)
+				um.GrantUpgrade(self, "parachute", this);
```

Comment:
It would be nice to un-hardcode the name of the upgrade and make it into a field in `ParachutableInfo`.

Also, you need the `UpgradeManager` twice, so it's probably already worth caching it in the constructor.